hedera-node/src/test/java/com/hedera/services/txns/schedule/ScheduleDeleteTransitionLogicTest.java

```diff
@@ -0,0 +1,98 @@
+import com.hederahashgraph.api.proto.java.ScheduleID;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+import proto.ScheduleDelete;
+
+import java.time.Instant;
+
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_SCHEDULE_ID;
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+@RunWith(JUnitPlatform.class)
+public class ScheduleDeleteTransitionLogicTest {
+    long thisSecond = 1_234_567L;
+    private Instant now = Instant.ofEpochSecond(thisSecond);
+    private OptionValidator validator;
+    private ScheduleStore store;
+    private HederaLedger ledger;
+    private PlatformTxnAccessor accessor;
+    private TransactionContext txnCtx;
+
+    private ScheduleID schedule = IdUtils.asSchedule("1.2.3");
+
+    private AccountID payer = IdUtils.asAccount("1.2.3");
+
+    private TransactionBody scheduleDeleteTxn;
+    private ScheduleDeleteTransitionLogic subject;
+
+    @BeforeEach
+    private void setup() {
+        validator = mock(OptionValidator.class);
+        store = mock(ScheduleStore.class);
+        ledger = mock(HederaLedger.class);
+        accessor = mock(PlatformTxnAccessor.class);
+
+        txnCtx = mock(TransactionContext.class);
+        given(txnCtx.activePayer()).willReturn(payer);
+        given(txnCtx.consensusTime()).willReturn(Instant.now());
+
+        subject = new ScheduleDeleteTransitionLogic(validator, store, ledger, txnCtx);
+    }
+
+    @Test
+    public void hasCorrectApplicability() {
+        givenValidTxnCtx();
+
+        // expect:
+        assertTrue(subject.applicability().test(scheduleDeleteTxn));
+        assertFalse(subject.applicability().test(TransactionBody.getDefaultInstance()));
+    }
+
+    @Test
+    public void failsOnInvalidSchedule() {
+        givenCtx(true);
+
+        // expect:
+        assertEquals(INVALID_SCHEDULE_ID, subject.validate(scheduleDeleteTxn));
+    }
+
+    private void givenValidTxnCtx() {
+        givenCtx(false);
+    }
+
+
+    private void givenCtx(
+            boolean invalidScheduleId
+    ) {
+        var builder = TransactionBody.newBuilder();
+        var scheduleDelete = ScheduleDelete.ScheduleDeleteTransactionBody.newBuilder()
+                .setSchedule(schedule);
+
+        if (invalidScheduleId) {
+            scheduleDelete.clearSchedule();
+        }
+
+        builder.setScheduleDelete(scheduleDelete);
+
+        scheduleDeleteTxn = builder.build();
+        given(accessor.getTxn()).willReturn(scheduleDeleteTxn);
+        given(txnCtx.accessor()).willReturn(accessor);
+        given(txnCtx.consensusTime()).willReturn(now);
```

Comment:
you don't need this

Reply:
I do need this `scheduleDeleteTxn = builder.build();`

Reply:
yes, you need it. I was mentioning `given(txnCtx.consensusTime()).willReturn(now);` only